server/repository/src/db_diesel/stock_line.rs

```diff
@@ -214,6 +217,16 @@ fn create_filtered_query(filter: Option<StockLineFilter>) -> BoxedStockLineQuery
                 .select(location_dsl::id.nullable());
             query = query.filter(stock_line_dsl::location_id.eq_any(location_ids));
         }
+
+        if master_list.is_some() {
+            let item_link_ids = MasterListLineRepository::create_filtered_query(Some(
+                MasterListLineFilter::new().master_list(master_list.unwrap()),
+            ))
+            .unwrap()
+            .select(master_list_line_dsl::item_link_id);
```

Comment:
If item is merged, master list stock line filter might not work, I think we should use item dsl here https://github.com/msupply-foundation/open-msupply/compare/4015-allow-filtering-view-stock-page-by-master-list...4015-allow-filtering-view-stock-page-by-master-list(use-Item-id-instead-if-item-lin-id)